cql3/functions/user_function.cc

```diff
@@ -56,9 +58,25 @@ bytes_opt user_function::execute(cql_serialization_format sf, const std::vector<
             }
             return lua::run_script(lua::bitcode_view{ctx.bitcode}, values, return_type(), ctx.cfg).get0();
         },
-        [&] (wasm::context& ctx) {
+        [&] (wasm::context& ctx) -> bytes_opt {
             try {
-                return wasm::run_script(ctx, arg_types(), parameters, return_type(), _called_on_null_input).get0();
+                auto func_cache = ctx.cache;
+                auto func_inst = func_cache->get(name(), ctx).get0();
+                std::exception_ptr ex;
+                bytes_opt ret;
+                try {
+                    ret = wasm::run_script(ctx, func_inst->store, func_inst->instance, func_inst->func, arg_types(), parameters, return_type(), _called_on_null_input).get0();
+                } catch (const wasm::instance_corrupting_exception& e) {
+                    func_cache->reset(name());
```

Comment:
Isn't only func_inst affected? If so, you shouldn't clear other entries. Just don't put this one in the cache.

Reply:
In the `reset()` implementation it is in fact only the instance changing, but we can do this without the method if we actually move instances when getting them